### ISSUE TITLE
`eseries_themoon_rce` check erroneously shows "vulnerable"

### ISSUE THREAD

**Issue by kdex** [threat9/routersploit#551]
## Steps to Reproduce (for bugs)
1. Grab a TP-Link TL-WR841N.
2. In `rsf`, run an `autopwn` scan against it.
3. Watch as `exploits/routers/linksys/eseries_themoon_rce` is erroneously detected to be `vulnerable`, despite it being a completely different router brand.
4. If you're unsure, try the blind RCE to verify that none of what you enter is really being executed.

## Your Environment
* RouterSploit Version used: AUR package, `3.4.0-1`
* Operating System and version: Arch Linux, `4.19.2-arch1-1-ARCH`
* Python Version: `3.7.1`
* Python Environment: N/A

## Current Behavior
* The exploit in question is a Linksys exploit; the TP-Link TL-WR841N is not vulnerable to it.
* In fact, all that occurs is a [check for a file on the HTTP server](https://github.com/threat9/routersploit/blob/master/routersploit/modules/exploits/routers/linksys/eseries_themoon_rce.py#L81-L87), which turns out to prove as quite unreliable.

## Expected Behavior
* The result for the TP-Link router should be `not vulnerable`.

**Comment by lucyoa** [comment]
It's not possible to uniquely verify exploitation of every vulnerability. By using blind command injection test approach we are going down the road of changing integrity of the device during scanning. 

RouterSploit is supposed to be an exploitation framework not a scanner (yes, we have modules that scan for vulnerabilities but its not what we are aiming for). We assume that people have certain level of knowledge and know what they are doing. 

The only thing that we can add is introducing some kind of assurance based on the quality of check method - does it really exploit specific vulnerability or just checks for the existence of specific file.

**Comment by kdex** [comment]
@lucyoa A reasonable change that I could imagine would be to extend scanners to return one state in the set:
```
  VULNERABLE,       // Verifiably vulnerable
  INVULNERABLE,     // Verifiably invulnerable
  MAYBE_VULNERABLE, // Conditions for vulnerability are met (e.g. file existance),
                    // though we can't verify without affecting the device's integrity
  UNKNOWN           // Requires manual fiddling
```

Though whether that is the simplest change or not is a different story. :)

**Comment by Keskebeu** [comment]
use exploits/routers/linksys/eseries_themoon_rce

Exploit failed to transfer payload

**Comment by Anton-Juul-Naber** [comment]
I have had the same issue with a bunch of different routers. The problem is just that soooo many devices will send back valid response headers, even though nothing is on that particular URL.

If there really isn't a better way to check for the vulnerablility, i think it should never be able to directly verify the vulnerabililty.
So if a valid response header comes back, it should simply return "Could not be verified" instead of "Is vulnerable".
That change would be very simple to implement, and would be a much more accurate description of what it actually checks for, at least in my opinion :)

**Comment by DHIRAL2908** [comment]
Yeah I have the same issue. It shows a Tenda router vulnerable... While I execute any command on CMD, it does nothing... I think its actually a false positive.

**Comment by OverlordAdmin** [comment]
Same issue scanning a Synology router.

**Comment by Lamera** [comment]
I have the same problem with a homemade router.

> cmd (MIPSLE Reverse TCP) > run
> [*] Using wget method
> [*] Using wget to download binary
> [-] Exploit failed to transfer payload

**Comment by imiddle** [comment]
Right now the `check` method for this exploit returns True whenever a GET request for `/tmUnblock.cgi` returns a status code of 200, 301, or 302.

Is there a more specific check that could be made against the text of the response, @lucyoa ?

My (not vulnerable) router returns status 200 with a meta refresh tag to its home page, so I can contribute the following patch which removes the false positive for my router but probably still allows a lot of other false positives:

```
--- a/routersploit/modules/exploits/routers/linksys/eseries_themoon_rce.py
+++ b/routersploit/modules/exploits/routers/linksys/eseries_themoon_rce.py
@@ -83,7 +83,13 @@
             path="/tmUnblock.cgi",
         )
 
-        if response and response.status_code in [200, 301, 302]:
+        if response is None:
+            return False
+
+        if '<meta http-equiv="refresh"' in response.text:
+            return False
+
+        if response.status_code in [200, 301, 302]:
             return True  # target is vulnerable
 
         return False  # target is not vulnerable
```

**Comment by gonotice-lang** [comment]
a useless toy, no matter how many tried to exploit the vulnerability, it can not find the server then failed to transfer payload, it is better to use msf

**Comment by xqxSpl0iT** [comment]
![sucks](https://user-images.githubusercontent.com/84131082/118303825-6e825100-b4d5-11eb-9163-0f011386700e.png)


The same issue was tested in 3 routers
